core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -81,6 +82,20 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if
+   * `spark.sql.catalog.spark_catalog` config is missing. We do not check for
+   * `spark.sql.extensions` because DeltaSparkSessionExtension can alternatively
+   * be activated using the `.withExtension()` API. This check can be disabled
+   * by setting DELTA_CHECK_REQUIRED_SPARK_CONF to false.
+   */
+  if(spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_CHECK_REQUIRED_SPARK_CONF)) {
```

Comment:
could you add a new method for this code such as:
```
/**
...
*/
protected def checkRequiredConfigurations() {
 if(spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_CHECK_REQUIRED_SPARK_CONF)) {
    if (spark.sparkContext.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
      throw DeltaErrors.configureSparkSessionWithExtensionAndCatalog(None)
    }
  }
}

checkRequiredConfigurations()
```

Reply:
done